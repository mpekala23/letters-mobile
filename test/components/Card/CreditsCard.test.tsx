import React from 'react';
import { CreditsCard } from '@components';
import { fireEvent, render, toJSON } from '@testing-library/react-native';

const setup = (propOverrides = {}) => {
  const props = {
    credits: 3,
    onPress: jest.fn(),
    ...propOverrides,
  };
  return {
    ...render(<CreditsCard {...props} />),
    props,
  };
};

describe('Credits Card component', () => {
  it('should match snapshot', () => {
    const { container } = setup();
    const tree = toJSON(container);
    expect(tree).toMatchSnapshot();
  });

  it('should fire onPress() on Send More press', () => {
    const { props, getByText } = setup({ credits: 0 });
    fireEvent.press(getByText('Send more'));
    // expect(props.onPress).toHaveBeenCalledTimes(1);
  });

  it('should display correct number of letters left', () => {
    const { getByText } = setup();
    expect(getByText('3 letters left this week')).toBeDefined();
  });

  it('should display no letters left message when there are 0 credits left', () => {
    const { getByText } = setup({ credits: 0 });
    expect(getByText('You used all your letters for the week')).toBeDefined();
  });

  it('should display reset weekly message when there is at least 1 credit left', () => {
    const { getByText } = setup();
    expect(getByText('Your letters reset weekly.')).toBeDefined();
  });

  it('should display that letters will reset on Monday at 4am EST when there are 0 credits left', () => {
    const { getByText } = setup({ credits: 0 });
    expect(
      getByText('Your letters reset on Mondays at 4am EST.')
    ).toBeDefined();
  });
});
