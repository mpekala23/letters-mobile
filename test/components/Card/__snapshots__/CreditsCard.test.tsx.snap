// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Credits Card component should match snapshot 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderRadius": 6,
          "marginVertical": 6,
          "padding": 15,
          "width": "100%",
        },
        Object {
          "elevation": 4,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.23,
          "shadowRadius": 2.62,
        },
        undefined,
      ]
    }
    testID="creditsCard"
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Poppins-Bold",
                },
                Object {
                  "color": "#282828",
                  "fontSize": 20,
                },
                Object {
                  "flex": 1,
                },
              ]
            }
          >
            4
             
            Credits
             
            left
          </Text>
          <TouchableOpacity
            activeOpacity={0.2}
          >
            <Text
              style={
                Object {
                  "color": "#762B2B",
                  "fontSize": 16,
                }
              }
            >
              Add more
            </Text>
          </TouchableOpacity>
        </View>
        <Text
          style={
            Object {
              "color": "#6E757C",
              "fontSize": 14,
            }
          }
        >
          Credits reset daily
        </Text>
      </View>
    </View>
  </View>
</View>
`;
